Modify workflow trigger to push on main (#1251)
opencontainers/image-spec: README.md

# OCI Image Format Specification

> [![Render and Lint Documentation](https://github.com/opencontainers/image-spec/actions/workflows/docs-and-linting.yml/badge.svg)](https://github.com/opencontainers/image-spec/actions/workflows/docs-and-linting.yml)

Author: ```suggestion
[![Render and Lint Documentation](https://github.com/opencontainers/image-spec/actions/workflows/docs-and-linting.yml/badge.svg?branch=main)](https://github.com/opencontainers/image-spec/actions/workflows/docs-and-linting.yml)
```

> [![Render and Lint Documentation](https://github.com/opencontainers/image-spec/actions/workflows/docs-and-linting.yml/badge.svg?branch=main)](https://github.com/opencontainers/image-spec/actions/workflows/docs-and-linting.yml)

Reviewer: If we're changing this (I would have also been fine with no change since shields.io seems a bit more flexible), lets pin to the push event:

```suggestion
[![Render and Lint Documentation](https://github.com/opencontainers/image-spec/actions/workflows/docs-and-linting.yml/badge.svg?branch=main&event=push)](https://github.com/opencontainers/image-spec/actions/workflows/docs-and-linting.yml)
```

Author: ```suggestion
![GitHub Actions for Docs and Linting](https://img.shields.io/github/actions/workflow/status/opencontainers/image-spec/docs-and-linting.yml?branch=main&label=GHA%20docs%20and%20linting)
```
